ui/components/app/confirm-page-container/confirm-page-container-navigation/confirm-page-container-navigation.component.js

```diff
@@ -42,7 +53,11 @@ const ConfirmPageContainerNavigation = () => {
   const onNextTx = (txId) => {
     if (txId) {
       dispatch(clearConfirmTransaction());
-      history.push(`${CONFIRM_TRANSACTION_ROUTE}/${txId}`);
+      history.push(
+        uncofirmedTransactions[txId]?.msgParams
+          ? `${CONFIRM_TRANSACTION_ROUTE}/${txId}${SIGNATURE_REQUEST_PATH}`
+          : `${CONFIRM_TRANSACTION_ROUTE}/${txId}`,
```

Comment:
I think if you route all request to `${CONFIRM_TRANSACTION_ROUTE}/${txId}` it takes care to re-route to specific page depending on request type and should work for any confirmation request in fact.